storage/namespace.go

```diff
@@ -268,7 +278,7 @@ func newDatabaseNamespace(
 		log:                    logger,
 		increasingIndex:        increasingIndex,
 		commitLogWriter:        commitLogWriter,
-		reverseIndex:           nil, // FOLLOWUP(prateek): will be done in https://github.com/m3db/m3db/pull/507
+		reverseIndex:           index,
```

Comment:
little weird to call this field reverse index when we dont use the word reverse anywhere else

Reply:
happy to rename, any suggestions?

Reply:
can we just call it `index`? I'm not opposed to calling it `reverseIndex` just seems like it should be consistent with whatever we're calling the index interface

Reply:
leaving as `reverseIndex` per the other discussion